packages/fast-tooling-react/src/css-property-editor/property-editor.tsx

```diff
@@ -0,0 +1,270 @@
+        valueWidth: number,
+        index: number
+    ): React.ReactNode {
+        return (
+            <div key={index}>
+                <input
+                    type={"text"}
+                    className={this.generateKeyClassNames()}
+                    onChange={this.handlePropertyKeyChange(key)}
+                    value={key}
+                    style={{ width: `${keyWidth}px` }}
+                />
+                :
+                <input
+                    type={"text"}
+                    className={this.generateValueClassNames()}
+                    onChange={this.handlePropertyValueChange(key)}
+                    value={value}
+                    style={{ width: `${valueWidth}px` }}
+                />
+                ;
+            </div>
+        );
+    }
+
+    private renderNewPropertyInput(
+        keyWidth: number,
+        valueWidth: number
+    ): React.ReactNode {
+        return (
+            <div>
+                <input
+                    type={"text"}
+                    className={this.generateKeyClassNames()}
+                    onChange={this.handleNewPropertyKeyChange}
+                    value={this.state.key}
+                    style={{ width: `${keyWidth}px` }}
+                    ref={this.keyInputRef}
+                />
+                :
+                <input
+                    type={"text"}
+                    className={this.generateValueClassNames()}
+                    onChange={this.handleNewPropertyValueChange}
+                    onKeyDown={this.handleNewPropertyValueKeyDown}
+                    value={this.state.value}
+                    style={{ width: `${valueWidth}px` }}
+                />
+                ;
+            </div>
+        );
+    }
+
+    private generateKeyClassNames(): string {
+        return `${this.props.managedClasses.cssPropertyEditor_input} ${
+            this.props.managedClasses.cssPropertyEditor_inputKey
+        }`;
+    }
+
+    private generateValueClassNames(): string {
+        return `${this.props.managedClasses.cssPropertyEditor_input} ${
+            this.props.managedClasses.cssPropertyEditor_inputValue
+        }`;
+    }
+
+    private getCSSPropertyConfig(data: CSSProperties): CSSPropertiesConfig {
+        const dataConfig: CSSPropertiesConfig = {};
+
+        if (!data) {
+            return dataConfig;
+        }
+
+        Object.keys(data).map(
+            (dataKey: string): void => {
+                dataConfig[dataKey] = {
+                    value: data[dataKey],
+                    keyWidth: 6.6 * dataKey.length,
+                    valueWidth: 6.6 * data[dataKey].length,
+                };
+            }
+        );
+
+        return dataConfig;
+    }
+
+    private handleClick = (e: React.MouseEvent<HTMLPreElement>): void => {
+        if ((e.target as HTMLElement).nodeName !== "INPUT") {
+            this.keyInputRef.current.focus();
+        }
+
+        e.stopPropagation();
```

Comment:
Why do you need to stop propagation?

Reply:
I don't, it's a hold over from a previous iteration, I'll remove.